### DISCUSSION TITLE
Guys, express your (dis)satisfaction!

### DISCUSSION THREAD

**Discussion by davidrapan** [davidrapan/ha-solarman#57] [Polls]
Be honest! 😆

**Comment by Linus-ui** [comment]
Davide,sei Grande.
L'integrazione Solarman,funziona Benissimo

**Comment by Edas66** [comment]
I have been using Stephan Joubert's integration for over a year, and since December 24, communication with the logger has been intermittent. Your integration looks stable, and has been working perfectly for several days. Thanks.
My inverter is Sofar Solar HYD 6KTL-3PH + battery Amass GTX3000-H

**Comment by Mira141** [comment]
Hi David, your work is super excellent, but I have a tip for further improvement for developers: I would like you to add a user parameter to set the communication refresh interval with the (DEYE) inverter directly as adjustable from the GUI in HA. Right now it is 5 seconds by default, but in case of debugging and editing HA it would be useful to be able to check the results independently of Solarman or Deyecloud, but there is a specific blockage at the WiFi dongle queue and thus a mismatch between the response of DeyCloud and HA. The possibility to set the com interval between 5 and 60 sec to balance both responses would be useful.

**Comment by musildavid** [comment]
Hi David,
I have a KEHUA iStorageE3 inverter which is not integrated. I also have the modbus communication protocol from this inverter. Unfortunately I am a beginner in HA. Would you be interested in programming the communication protocol with Solarman? Please offer a price for the integration. Thank you David

**Reply by davidrapan** [reply]
Hi @musildavid, if you can provide documentation of the modbus communication protocol for your device, I can take a look at it in the near future. 😉

**Reply by musildavid** [reply]
Hi @davidrapan ,
thank you very much for your response. I am attaching the documentation for my iStorage E3. Thank you, best regards, David
[WRWF-0I003-02  Energy Storage System iStoragE Series Northbound Monitoring Modbus Communication Protocol (V1.04) 001 231011.pdf](https://github.com/user-attachments/files/19591371/WRWF-0I003-02.Energy.Storage.System.iStoragE.Series.Northbound.Monitoring.Modbus.Communication.Protocol.V1.04.001.231011.pdf)

**Reply by musildavid** [reply]
I use a data logger Solarman LSW-3

**Reply by musildavid** [reply]
Hello @daviddrapan,
I would like to remind myself with an edit. Do you have time?

**Reply by davidrapan** [reply]
Hello @musildavid, what do you need me for? :)

**Reply by musildavid** [reply]
Hi David,
I have a KEHUA iStorageE3 inverter which is not integrated. I also have the modbus communication protocol from this inverter. Unfortunately I am a beginner in HA. Would you be interested in programming the communication protocol with Solarman? Please offer a price for the integration. Thank you David

**Reply by musildavid** [reply]
[WRWF-0I003-02.Energy.Storage.System.iStoragE.Series.Northbound.Monitoring.Modbus.Communication.Protocol.V1.04.001.231011.pdf](https://github.com/user-attachments/files/20187294/WRWF-0I003-02.Energy.Storage.System.iStoragE.Series.Northbound.Monitoring.Modbus.Communication.Protocol.V1.04.001.231011.pdf)

**Reply by davidrapan** [reply]
I'll see if I have some free time later this week., 😉

**Comment by Mira141** [comment]
Hi David Musil, 
if not succed with David Rapan (i suppose he is busy), try simply take DEYE 3p yaml interver definition file, copy it under name KEHUAiStorageE3 and edit it direct in HA file editor (or other external editor, I use free Notepad++ in Windows or Kate in Linux), compare line per line the register addressing according to differences in Modbus protocols of both and make corrections. Modbus of DEYE is downloadable from David´s links ... When finish, simply edit solarman integration settings to manually point at your new inverter definition file and it should work.  Maybe some functions of your inverter will not correspond to DEYE (or other inverter), then some sensors will be not displayed. Don´t expect, that it will work 1:1 , but after some playing with HA you will find how things works. It took me (beginner in December 2024) about 15 hours of learning to integrate inverter and make nice dashboard ... Similar way I used to remake nice PV Dashboard from Goodwe at Deye. Most entities have to be renamed, but result is OK ...  I am sure, you want it too  :-) 
![2025-14-01-PVdashboard](https://github.com/user-attachments/assets/0b7c02df-da0e-4030-a52b-7977375447b9)

**Reply by davidrapan** [reply]
Hey @Mira141, you can share your inverter profile if you want. 😉

**Reply by Mira141** [reply]
Hi, here is yaml of DEYE dashboard for HA Solarman Integration. It´s still under development, not finished, not aligned to borders, but works as study example both at PC and mobile/tablet app ... and by debugging you can change it and get opinion how it works by creating your own ... It´s, particulary translated to Czech language, but everybody can edit it according to own needs ... Take it as is made, as study example for your starting point.  No complains, no wishes to customize. You will make it self  :-)  Maybe you will need to add some integrations thru HACS to get it work ... I wil post list of my ones....   
EDIT: Direct listing distorted formatting and cannot be direct copied to HA . Therefore i replaced with zipped export from Notepad++ ...  

[DEYE-Dashboard.zip](https://github.com/user-attachments/files/20232191/DEYE-Dashboard.zip)

**Reply by Mira141** [reply]
Not all are needed for dashboard ... no Node-Red, no influx DB ... but are used under automations development ... This amount of integrations takes about 30% of memory and disk and 5% of CPU at HA Green...   
![HACS-integrations](https://github.com/user-attachments/assets/27378f96-e9c3-4959-812d-00100831942b)
![HA-add-on](https://github.com/user-attachments/assets/68302963-fe76-4b77-af68-d3fa87ac6c97)

**Reply by Mira141** [reply]
Maybe you will find this HA daily graph useful instead of Solarman and Deyecloud history. Not finished yet, target is kW unit at left y axe to save place at mobiles/tablet devices. Similar the white button with unreadable yellow text at right bottom corner is very wierd alien entity, that occured out of sight :-D  ... Take as study case ...

EDIT: I see that this page distorts formatting in direct listing, therefore second attempt to export yaml file from Notepad++. This has to keep formatting ... must be zipped to upload... 
[DailyChart.zip](https://github.com/user-attachments/files/20232168/DailyChart.zip)

![obrazek](https://github.com/user-attachments/assets/26a4082d-9428-42ec-89b2-7aa7984828ca)

**Reply by davidrapan** [reply]
Hey @Mira141, I was also talking about your inverter profile that you are using in this integration.

Thanks, when I have some time, I'll add it to the wiki. 😉

**Comment by Mira141** [comment]
Hi David Rapan, totally satisfied with your work, excellent (y)  ,
 but I  have am important wish:
 Please explain me the entities controlling the "Grid peak-shaving" and "Gen peak-shaving" switches in DEYE 12 kW SG04-EU, because due to the incorrect sales control logic from DEYE, Gen peak-shaving has the effect of prohibiting sales, even when Selling First is active in Work mode 1. If they are not in the 158 entities of the current version, please read them from the inverter and control them from HA and add them to the list of entities. This is necessary for the correct control of the energy spot exchange. Thanks in advance ...

**Reply by davidrapan** [reply]
Hi @Mira141, those entities have been added into the code already and will be part of next release.

If you don't want to wait you can manually update whole file or add specific entities you are missing from [deye_p3.yaml](https://github.com/davidrapan/ha-solarman/blob/main/custom_components/solarman/inverter_definitions/deye_p3.yaml) to the file in HA at `/config/custom_components/solarman/inverter_definitions/deye_p3.yaml`

**Reply by Mira141** [reply]
**Many thanks**, new entities will be used in developed Node-RED automations of HA and DEYE spot energy exchange ...

**Comment by marcofulco1982** [comment]
Hi David,
i have a Hiconics HEC2-S6.0Hr2 with an LSW-5 data logger. i can get connected but i cant see any values. i don't know the register map and id love to get this integrated into home assistant so i can complete my dashboard.
do you have any answers.
Thanks

**Reply by davidrapan** [reply]
Hi @marcofulco1982, try contacting the manufacturer and ask for Modbus RTU documentantion.

**Comment by marcofulco1982** [comment]
Thanks David. I have asked but nothing yet. Also can I ask something? When home assistant is connected does it take realtime instantaneous values from the inverter or is it still the same collected data you see in the app?I know that the app data is only collected every 5 minutes so it’s more of just a screenshot. Is there a way to pull the actual real time data?RegardsMarco Fulco On 20 Nov 2025, at 20:43, David Rapan ***@***.***> wrote:﻿
Hi @marcofulco1982, try contacting the manufacturer and ask for Modbus RTU documentantion.

**Reply by davidrapan** [reply]
The integration fetches real-time data every 5 seconds.

**Comment by marcofulco1982** [comment]
Thanks David. Now I just have to get the mobs rtu docs. RegardsMarco Fulco On 21 Nov 2025, at 08:30, David Rapan ***@***.***> wrote:﻿
The integration fetches real-time data every 5 seconds.
